lib/install.js

```diff
@@ -31,34 +34,29 @@ function install (gyp, argv, callback) {
     if (err) {
       log.warn('install', 'got an error, rolling back install')
       // roll-back the install if anything went wrong
-      gyp.commands.remove([ version ], function (err2) {
+      gyp.commands.remove([ release.versionStr ], function (err2) {
         callback(err)
       })
     } else {
-      callback(null, version)
+      callback(null, release.versionStr)
     }
   }
 
-  var distUrl = gyp.opts['dist-url'] || gyp.opts.disturl || 'https://nodejs.org/dist'
-
-
   // Determine which node dev files version we are installing
-  var versionStr = argv[0] || gyp.opts.target || process.version
-  log.verbose('install', 'input version string %j', versionStr)
+  log.verbose('install', 'input version string %j', release.versionStr)
 
   // parse the version to normalize and ensure it's valid
-  var version = semver.parse(versionStr)
-  if (!version) {
-    return callback(new Error('Invalid version number: ' + versionStr))
+  if (!release.versionSemver) {
+    return callback(new Error('Invalid version number: ' + release.versionStr))
   }
 
-  if (semver.lt(versionStr, '0.8.0')) {
-    return callback(new Error('Minimum target version is `0.8.0` or greater. Got: ' + versionStr))
+  if (semver.lt(release.versionStr, '0.8.0')) {
+    return callback(new Error('Minimum target version is `0.8.0` or greater. Got: ' + release.versionStr))
   }
 
   // 0.x.y-pre versions are not published yet and cannot be installed. Bail.
-  if (version.prerelease[0] === 'pre') {
-    log.verbose('detected "pre" node version', versionStr)
+  if (release.versionSemver.prerelease[0] === 'pre') {
+    log.verbose('detected "pre" node version', release.versionStr)
```

Comment:
see my other dangling comment about `-pre` coming from self-compiled-from-repository builds, also you can `npm config set nodedir=/path/to/repo` and also set the `npm_config_nodedir=/path/to/repo` for the same effect